{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b7f7e56b-de81-4160-b42b-2a33fe1265c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Attaching SeuratObject\n",
      "\n",
      "Attaching sp\n",
      "\n",
      "Registered S3 method overwritten by 'SeuratDisk':\n",
      "  method            from  \n",
      "  as.sparse.H5Group Seurat\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(Seurat)\n",
    "library(SeuratDisk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2bb05423-418d-4877-8c30-456db1f863bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna = readRDS('/home/icb/zihe.zheng/projects/microglia/data/SUN/ROSMAP.ImmuneCells.6regions.snRNAseq.counts.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "060ea573-b71d-463f-b85a-1c8f46078d98",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta = readRDS('/home/icb/zihe.zheng/projects/microglia/data/SUN/ROSMAP.ImmuneCells.6regions.snRNAseq.meta.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3726725d-0e86-418a-8ed9-df9f9e02c287",
   "metadata": {},
   "outputs": [],
   "source": [
    "atac = readRDS('/home/icb/zihe.zheng/projects/microglia/data/SUN/microglia.snATAC.ArchRobj.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "38b4bbe5-c7c4-41ab-bcf5-bbde0ba552b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "seurat_object <- CreateSeuratObject(rna, project = \"SeuratProject\", assay = \"RNA\", meta.data = meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "34c1f5eb-ac20-48d4-ba5e-6629b78de4ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "seurat_object[[\"RNA\"]] <- as(object = seurat_object[[\"RNA\"]], Class = \"Assay\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "25cbea84-3e7f-4ecc-8c7f-b7d7f9d05b15",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Creating h5Seurat file for version 3.1.5.9900\n",
      "\n",
      "Adding counts for RNA\n",
      "\n",
      "Adding data for RNA\n",
      "\n",
      "No variable features found for RNA\n",
      "\n",
      "No feature-level metadata found for RNA\n",
      "\n"
     ]
    }
   ],
   "source": [
    "SaveH5Seurat(seurat_object, filename = \"/home/icb/zihe.zheng/projects/microglia/data/SUN/MIT_ROSMAP_RNA.h5Seurat\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b7f231ea-1dbb-4c4a-8907-25dfd97c9bb9",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Validating h5Seurat file\n",
      "\n",
      "Initializing RNA with data\n",
      "\n",
      "Adding counts for RNA\n",
      "\n",
      "Adding miscellaneous information for RNA\n",
      "\n",
      "Adding command information\n",
      "\n",
      "Adding cell-level metadata\n",
      "\n",
      "Adding miscellaneous information\n",
      "\n",
      "Adding tool-specific results\n",
      "\n"
     ]
    }
   ],
   "source": [
    "seurat_object_s <- LoadH5Seurat(\"/home/icb/zihe.zheng/projects/microglia/data/SUN/MIT_ROSMAP_RNA.h5Seurat\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3d05cf90-0ec0-413b-b609-a152194fc389",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Validating h5Seurat file\n",
      "\n",
      "Adding data from RNA as X\n",
      "\n",
      "Adding counts from RNA as raw\n",
      "\n",
      "Transfering meta.data to obs\n",
      "\n"
     ]
    }
   ],
   "source": [
    "Convert(\"/home/icb/zihe.zheng/projects/microglia/data/SUN/MIT_ROSMAP_RNA.h5Seurat\", dest = \"h5ad\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfb08f1b-9371-4a60-96e4-56d916f77224",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5afee17-e614-4802-87e0-db45547b8eb0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e8a3e229-b035-4f2e-a35e-b34ea3a2a64e",
   "metadata": {},
   "outputs": [],
   "source": [
    "if (!requireNamespace(\"BiocManager\", quietly = TRUE)) install.packages(\"BiocManager\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "73e545d0-62d9-4b9a-8718-cbd9a4e5e412",
   "metadata": {},
   "outputs": [],
   "source": [
    "if (!requireNamespace(\"devtools\", quietly = TRUE)) install.packages(\"devtools\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9dd8d0e0-7f55-4f82-baae-4e420365e310",
   "metadata": {},
   "source": [
    "## read atac-seq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "d8e74178-6038-427d-ac2f-4dfcd73dc2ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "Sys.setenv(CONDA_BUILD_SYSROOT=\"/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "77d178b8-9bec-480c-8229-2e2fd5368219",
   "metadata": {},
   "outputs": [],
   "source": [
    "options(buildtools.check = function(action) TRUE )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "62683694-4369-4dd0-a5a0-6c6eef37cb77",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using github PAT from envvar GITHUB_PAT\n",
      "\n",
      "Downloading GitHub repo GreenleafLab/ArchR@master\n",
      "\n",
      "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'help(\"repositories\", package = \"BiocManager\")' for details.\n",
      "Replacement repositories:\n",
      "    CRAN: https://cran.r-project.org\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "crayon       (1.5.1      -> 1.5.3    ) [CRAN]\n",
      "GlobalOpt... (NA         -> 0.1.2    ) [CRAN]\n",
      "colorspace   (2.0-3      -> 2.1-1    ) [CRAN]\n",
      "shape        (NA         -> 1.4.6.1  ) [CRAN]\n",
      "zlibbioc     (1.34.0     -> 1.36.0   ) [CRAN]\n",
      "IRanges      (2.22.1     -> 2.24.1   ) [CRAN]\n",
      "S4Vectors    (0.26.0     -> 0.28.1   ) [CRAN]\n",
      "BiocGenerics (0.34.0     -> 0.36.1   ) [CRAN]\n",
      "BH           (1.78.0-0   -> 1.84.0-0 ) [CRAN]\n",
      "Rcpp         (1.0.9      -> 1.0.13   ) [CRAN]\n",
      "RcppEigen    (0.3.3.9.2  -> 0.3.4.0.2) [CRAN]\n",
      "fs           (1.5.2      -> 1.6.4    ) [CRAN]\n",
      "glue         (1.6.2      -> 1.7.0    ) [CRAN]\n",
      "cli          (3.4.1      -> 3.6.3    ) [CRAN]\n",
      "sass         (0.4.2      -> 0.4.9    ) [CRAN]\n",
      "lifecycle    (1.0.2      -> 1.0.4    ) [CRAN]\n",
      "cachem       (1.0.6      -> 1.1.0    ) [CRAN]\n",
      "tinytex      (0.42       -> 0.52     ) [CRAN]\n",
      "fontawesome  (0.3.0      -> 0.5.2    ) [CRAN]\n",
      "bslib        (0.4.0      -> 0.8.0    ) [CRAN]\n",
      "xfun         (0.33       -> 0.47     ) [CRAN]\n",
      "highr        (0.9        -> 0.11     ) [CRAN]\n",
      "evaluate     (0.16       -> 0.24.0   ) [CRAN]\n",
      "later        (1.2.0      -> 1.3.2    ) [CRAN]\n",
      "yaml         (2.3.5      -> 2.3.10   ) [CRAN]\n",
      "rmarkdown    (2.16       -> 2.28     ) [CRAN]\n",
      "knitr        (1.40       -> 1.48     ) [CRAN]\n",
      "rlang        (1.0.6      -> 1.1.4    ) [CRAN]\n",
      "fastmap      (1.1.0      -> 1.2.0    ) [CRAN]\n",
      "digest       (0.6.29     -> 0.6.37   ) [CRAN]\n",
      "promises     (1.2.0.1    -> 1.3.0    ) [CRAN]\n",
      "crosstalk    (1.2.0      -> 1.2.1    ) [CRAN]\n",
      "jsonlite     (1.8.0      -> 1.8.8    ) [CRAN]\n",
      "httpuv       (1.6.6      -> 1.6.15   ) [CRAN]\n",
      "htmlwidgets  (1.5.4      -> 1.6.4    ) [CRAN]\n",
      "htmltools    (0.5.3      -> 0.5.8.1  ) [CRAN]\n",
      "commonmark   (1.8.0      -> 1.9.1    ) [CRAN]\n",
      "withr        (2.5.0      -> 3.0.1    ) [CRAN]\n",
      "sourcetools  (0.1.7      -> 0.1.7-1  ) [CRAN]\n",
      "shiny        (1.7.2      -> 1.9.1    ) [CRAN]\n",
      "sys          (3.4        -> 3.4.2    ) [CRAN]\n",
      "stringi      (1.7.8      -> 1.8.4    ) [CRAN]\n",
      "utf8         (1.2.2      -> 1.2.4    ) [CRAN]\n",
      "askpass      (1.1        -> 1.2.0    ) [CRAN]\n",
      "cpp11        (0.4.2      -> 0.4.7    ) [CRAN]\n",
      "stringr      (1.4.1      -> 1.5.1    ) [CRAN]\n",
      "fansi        (1.0.3      -> 1.0.6    ) [CRAN]\n",
      "munsell      (0.5.0      -> 0.5.1    ) [CRAN]\n",
      "labeling     (0.4.2      -> 0.4.3    ) [CRAN]\n",
      "farver       (2.1.1      -> 2.1.2    ) [CRAN]\n",
      "openssl      (2.0.3      -> 2.2.1    ) [CRAN]\n",
      "curl         (4.3.2      -> 5.2.2    ) [CRAN]\n",
      "isoband      (0.2.5      -> 0.2.7    ) [CRAN]\n",
      "gtable       (0.3.1      -> 0.3.5    ) [CRAN]\n",
      "tidyselect   (1.1.2      -> 1.2.1    ) [CRAN]\n",
      "pillar       (1.8.1      -> 1.9.0    ) [CRAN]\n",
      "data.table   (1.14.2     -> 1.15.4   ) [CRAN]\n",
      "purrr        (0.3.4      -> 1.0.2    ) [CRAN]\n",
      "tibble       (3.1.8      -> 3.2.1    ) [CRAN]\n",
      "vctrs        (0.4.1      -> 0.6.5    ) [CRAN]\n",
      "dplyr        (1.0.10     -> 1.1.4    ) [CRAN]\n",
      "tidyr        (1.2.1      -> 1.3.1    ) [CRAN]\n",
      "viridisLite  (0.4.1      -> 0.4.2    ) [CRAN]\n",
      "httr         (1.4.4      -> 1.4.7    ) [CRAN]\n",
      "scales       (1.2.1      -> 1.3.0    ) [CRAN]\n",
      "ggplot2      (3.3.6      -> 3.5.1    ) [CRAN]\n",
      "formatR      (NA         -> 1.14     ) [CRAN]\n",
      "futile.op... (NA         -> 1.0.1    ) [CRAN]\n",
      "lambda.r     (NA         -> 1.2.4    ) [CRAN]\n",
      "snow         (NA         -> 0.4-4    ) [CRAN]\n",
      "futile.lo... (NA         -> 1.4.3    ) [CRAN]\n",
      "RCurl        (1.98-1.8   -> 1.98-1.16) [CRAN]\n",
      "Rhtslib      (NA         -> 1.22.0   ) [CRAN]\n",
      "BiocParallel (NA         -> 1.24.1   ) [CRAN]\n",
      "bitops       (1.0-7      -> 1.0-8    ) [CRAN]\n",
      "XVector      (0.28.0     -> 0.30.0   ) [CRAN]\n",
      "Biostrings   (NA         -> 2.58.0   ) [CRAN]\n",
      "GenomicRa... (1.40.0     -> 1.42.0   ) [CRAN]\n",
      "GenomeInfoDb (1.24.0     -> 1.26.7   ) [CRAN]\n",
      "matrixStats  (0.62.0     -> 1.3.0    ) [CRAN]\n",
      "DelayedArray (0.14.0     -> 0.16.3   ) [CRAN]\n",
      "Biobase      (2.48.0     -> 2.50.0   ) [CRAN]\n",
      "MatrixGen... (NA         -> 1.2.1    ) [CRAN]\n",
      "Summarize... (1.18.1     -> 1.20.0   ) [CRAN]\n",
      "GenomicAl... (NA         -> 1.26.0   ) [CRAN]\n",
      "XML          (NA         -> 3.99-0.17) [CRAN]\n",
      "Rsamtools    (NA         -> 2.6.0    ) [CRAN]\n",
      "rtracklayer  (NA         -> 1.50.0   ) [CRAN]\n",
      "prettyunits  (1.1.1      -> 1.2.0    ) [CRAN]\n",
      "plyr         (1.8.7      -> 1.8.9    ) [CRAN]\n",
      "tzdb         (NA         -> 0.4.0    ) [CRAN]\n",
      "vroom        (NA         -> 1.6.5    ) [CRAN]\n",
      "R.methodsS3  (NA         -> 1.8.2    ) [CRAN]\n",
      "R.oo         (NA         -> 1.26.0   ) [CRAN]\n",
      "pracma       (NA         -> 2.4.4    ) [CRAN]\n",
      "png          (0.1-7      -> 0.1-8    ) [CRAN]\n",
      "bit          (4.0.4      -> 4.0.5    ) [CRAN]\n",
      "Annotatio... (NA         -> 1.52.0   ) [CRAN]\n",
      "plogr        (NA         -> 0.2.0    ) [CRAN]\n",
      "blob         (NA         -> 1.2.4    ) [CRAN]\n",
      "KEGGREST     (NA         -> 1.30.1   ) [CRAN]\n",
      "R.utils      (NA         -> 2.12.3   ) [CRAN]\n",
      "GO.db        (NA         -> 3.12.1   ) [CRAN]\n",
      "annotate     (NA         -> 1.68.0   ) [CRAN]\n",
      "poweRlaw     (NA         -> 0.80.0   ) [CRAN]\n",
      "readr        (NA         -> 2.1.5    ) [CRAN]\n",
      "TFMPvalue    (NA         -> 0.0.9    ) [CRAN]\n",
      "seqLogo      (NA         -> 1.56.0   ) [CRAN]\n",
      "RSQLite      (NA         -> 2.3.7    ) [CRAN]\n",
      "DBI          (NA         -> 1.2.3    ) [CRAN]\n",
      "gtools       (3.9.3      -> 3.9.5    ) [CRAN]\n",
      "Dirichlet... (NA         -> 1.32.0   ) [CRAN]\n",
      "CNEr         (NA         -> 1.26.0   ) [CRAN]\n",
      "BSgenome     (NA         -> 1.58.0   ) [CRAN]\n",
      "Rhdf5lib     (1.10.0     -> 1.12.1   ) [CRAN]\n",
      "Cairo        (NA         -> 1.6-2    ) [CRAN]\n",
      "clue         (NA         -> 0.3-65   ) [CRAN]\n",
      "GetoptLong   (NA         -> 1.0.5    ) [CRAN]\n",
      "circlize     (NA         -> 0.4.16   ) [CRAN]\n",
      "dqrng        (0.3.0      -> 0.4.1    ) [CRAN]\n",
      "RSpectra     (0.16-1     -> 0.16-2   ) [CRAN]\n",
      "RcppAnnoy    (0.0.19     -> 0.0.22   ) [CRAN]\n",
      "irlba        (2.3.5      -> 2.3.5.1  ) [CRAN]\n",
      "FNN          (1.1.3.1    -> 1.1.4    ) [CRAN]\n",
      "RcppArmad... (0.11.1.1.0 -> 14.0.0-1 ) [CRAN]\n",
      "Rtsne        (0.16       -> 0.17     ) [CRAN]\n",
      "DT           (NA         -> 0.33     ) [CRAN]\n",
      "plotly       (4.10.0     -> 4.10.4   ) [CRAN]\n",
      "TFBSTools    (NA         -> 1.28.0   ) [CRAN]\n",
      "nabor        (NA         -> 0.5.0    ) [CRAN]\n",
      "rhdf5filters (NA         -> 1.2.1    ) [CRAN]\n",
      "ComplexHe... (NA         -> 2.6.2    ) [CRAN]\n",
      "ggrepel      (0.9.1      -> 0.9.5    ) [CRAN]\n",
      "uwot         (0.1.14     -> 0.2.2    ) [CRAN]\n",
      "chromVAR     (NA         -> 1.12.0   ) [CRAN]\n",
      "motifmatchr  (NA         -> 1.12.0   ) [CRAN]\n",
      "rhdf5        (2.32.0     -> 2.34.0   ) [CRAN]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Skipping 1 packages not available: rjson\n",
      "\n",
      "Installing 138 packages: crayon, GlobalOptions, rjson, colorspace, shape, zlibbioc, IRanges, S4Vectors, BiocGenerics, BH, Rcpp, RcppEigen, fs, glue, cli, sass, lifecycle, cachem, tinytex, fontawesome, bslib, xfun, highr, evaluate, later, yaml, rmarkdown, knitr, rlang, fastmap, digest, promises, crosstalk, jsonlite, httpuv, htmlwidgets, htmltools, commonmark, withr, sourcetools, shiny, sys, stringi, utf8, askpass, cpp11, stringr, fansi, munsell, labeling, farver, openssl, curl, isoband, gtable, tidyselect, pillar, data.table, purrr, tibble, vctrs, dplyr, tidyr, viridisLite, httr, scales, ggplot2, formatR, futile.options, lambda.r, snow, futile.logger, RCurl, Rhtslib, BiocParallel, bitops, XVector, Biostrings, GenomicRanges, GenomeInfoDb, matrixStats, DelayedArray, Biobase, MatrixGenerics, SummarizedExperiment, GenomicAlignments, XML, Rsamtools, rtracklayer, prettyunits, plyr, tzdb, vroom, R.methodsS3, R.oo, pracma, png, bit, AnnotationDbi, plogr, blob, KEGGREST, R.utils, GO.db, annotate, poweRlaw, readr, TFMPvalue, seqLogo, RSQLite, DBI, gtools, DirichletMultinomial, CNEr, BSgenome, Rhdf5lib, Cairo, clue, GetoptLong, circlize, dqrng, RSpectra, RcppAnnoy, irlba, FNN, RcppArmadillo, Rtsne, DT, plotly, TFBSTools, nabor, rhdf5filters, ComplexHeatmap, ggrepel, uwot, chromVAR, motifmatchr, rhdf5\n",
      "\n",
      "Warning message:\n",
      "“package ‘rjson’ is not available for this version of R\n",
      "‘rjson’ version 0.2.22 is in the repositories but depends on R (>= 4.4.0)\n",
      "\n",
      "A version of this package for your version of R might be available elsewhere,\n",
      "see the ideas at\n",
      "https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages”\n",
      "Warning message:\n",
      "“dependency ‘rjson’ is not available”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘colorspace’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘zlibbioc’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘Rcpp’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘fs’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘glue’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘cli’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘xfun’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘yaml’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘rlang’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘fastmap’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘digest’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘jsonlite’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘commonmark’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘sourcetools’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘sys’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘stringi’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘utf8’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘fansi’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘farver’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘curl’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘isoband’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘data.table’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘bitops’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘matrixStats’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘XML’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘png’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘bit’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘gtools’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘Rhdf5lib’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘Cairo’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘clue’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘irlba’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘FNN’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘S4Vectors’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘RcppEigen’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘lifecycle’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘cachem’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘later’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘htmltools’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘askpass’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘RCurl’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘Rhtslib’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘Biobase’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘plyr’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘tzdb’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘TFMPvalue’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘GetoptLong’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘dqrng’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘RcppAnnoy’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘RcppArmadillo’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘Rtsne’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘rhdf5filters’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘IRanges’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘sass’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘knitr’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘promises’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘openssl’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘gtable’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘vctrs’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘RSpectra’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘nabor’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘rhdf5’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘bslib’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘httpuv’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘stringr’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘pillar’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘purrr’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘BiocParallel’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘XVector’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘DelayedArray’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘DirichletMultinomial’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘ComplexHeatmap’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘uwot’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘rmarkdown’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘shiny’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘tibble’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘Biostrings’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘GenomicRanges’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘RSQLite’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘htmlwidgets’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘dplyr’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘ggplot2’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘SummarizedExperiment’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘Rsamtools’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘vroom’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘AnnotationDbi’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘KEGGREST’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘tidyr’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘GenomicAlignments’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘GO.db’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘annotate’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘readr’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘ggrepel’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘rtracklayer’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘CNEr’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘BSgenome’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘TFBSTools’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘chromVAR’ had non-zero exit status”\n",
      "Warning message in i.p(...):\n",
      "“installation of package ‘motifmatchr’ had non-zero exit status”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "* checking for file ‘/tmp/Rtmp5maGyq/remotes3206f055d36677/GreenleafLab-ArchR-d9e741c/DESCRIPTION’ ... OK\n",
      "* preparing ‘ArchR’:\n",
      "* checking DESCRIPTION meta-information ... OK\n",
      "* cleaning src\n",
      "* checking for LF line-endings in source and make files and shell scripts\n",
      "* checking for empty or unneeded directories\n",
      "* building ‘ArchR_1.0.2.tar.gz’\n",
      "Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :\n",
      "  storing paths of more than 100 bytes is not portable:\n",
      "  ‘ArchR/.github/ISSUE_TEMPLATE/bug-error-report---no-usage-questions---feature-requests--use-discussions--.md’\n",
      "Warning: invalid uid value replaced by that for user 'nobody'\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in i.p(...):\n",
      "“installation of package ‘/tmp/Rtmp5maGyq/file3206f0384a6465/ArchR_1.0.2.tar.gz’ had non-zero exit status”\n"
     ]
    }
   ],
   "source": [
    "devtools::install_github(\"GreenleafLab/ArchR\", ref=\"master\", repos = BiocManager::repositories())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e74aa35f-9c96-4af6-9f00-4560fd025087",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n",
      "ArchR : Version 1.0.2\n",
      "For more information see our website : www.ArchRProject.com\n",
      "If you encounter a bug please report : https://github.com/GreenleafLab/ArchR/issues\n",
      "\n",
      "Loading Required Packages...\n",
      "\n",
      "\tLoading Package : grid v4.4.1\n",
      "\n",
      "\tLoading Package : gridExtra v2.3\n",
      "\n",
      "\tLoading Package : gtools v3.9.5\n",
      "\n",
      "\tLoading Package : gtable v0.3.5\n",
      "\n",
      "\tLoading Package : ggplot2 v3.5.1\n",
      "\n",
      "\tLoading Package : magrittr v2.0.3\n",
      "\n",
      "\tLoading Package : plyr v1.8.9\n",
      "\n",
      "\tLoading Package : stringr v1.5.1\n",
      "\n",
      "\tLoading Package : data.table v1.16.0\n",
      "\n",
      "\tLoading Package : matrixStats v1.3.0\n",
      "\n",
      "\tLoading Package : S4Vectors v0.42.1\n",
      "\n",
      "\tLoading Package : GenomicRanges v1.56.1\n",
      "\n",
      "\tLoading Package : BiocGenerics v0.50.0\n",
      "\n",
      "\tLoading Package : Matrix v1.7.0\n",
      "\n",
      "\tLoading Package : Rcpp v1.0.13\n",
      "\n",
      "\tLoading Package : SummarizedExperiment v1.34.0\n",
      "\n",
      "\tLoading Package : rhdf5 v2.48.0\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(ArchR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6ed14c59-9dcd-4d20-ab95-b8c53c52c59a",
   "metadata": {},
   "outputs": [],
   "source": [
    "atac = readRDS('/home/icb/zihe.zheng/projects/microglia/data/SUN/microglia.snATAC.ArchRobj.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2d16a409-18c3-467d-ad2b-cbf09aea9a46",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "class: ArchRProject \n",
       "outputDirectory: /net/bmc-lab5/data/kellis/users/nasun/projects/project0.microglia/analysis.snATAC/1archr.luria.Mar2022/brain.microglia.filter \n",
       "samples(293): D19-122585 D19-122586 ... D19-8636 D19-8637\n",
       "sampleColData names(1): ArrowFiles\n",
       "cellColData names(19): subject Sample ... pmi cellstate\n",
       "numberOfCells(1): 41832\n",
       "medianTSS(1): 9.4885\n",
       "medianFrags(1): 4755.5"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "atac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "277ff448-f516-4342-abd2-4ea81abaa805",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50f5ea63-f507-425a-9002-dce36b225ebe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0ced7d1c-5274-4bc4-b8fc-ab43214138da",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in normalizePath(getOutputDirectory(ArchRProj)):\n",
      "“path[1]=\"/net/bmc-lab5/data/kellis/users/nasun/projects/project0.microglia/analysis.snATAC/1archr.luria.Mar2022/brain.microglia.filter\": No such file or directory”\n",
      "Copying ArchRProject to new outputDirectory : /ictstr01/home/icb/zihe.zheng/projects/microglia/data/SUN/arrows\n",
      "\n",
      "Copying Arrow Files...\n",
      "\n",
      "Copying Arrow Files (1 of 293)\n",
      "\n",
      "Copying Arrow Files (2 of 293)\n",
      "\n",
      "Copying Arrow Files (3 of 293)\n",
      "\n",
      "Copying Arrow Files (4 of 293)\n",
      "\n",
      "Copying Arrow Files (5 of 293)\n",
      "\n",
      "Copying Arrow Files (6 of 293)\n",
      "\n",
      "Copying Arrow Files (7 of 293)\n",
      "\n",
      "Copying Arrow Files (8 of 293)\n",
      "\n",
      "Copying Arrow Files (9 of 293)\n",
      "\n",
      "Copying Arrow Files (10 of 293)\n",
      "\n",
      "Copying Arrow Files (11 of 293)\n",
      "\n",
      "Copying Arrow Files (12 of 293)\n",
      "\n",
      "Copying Arrow Files (13 of 293)\n",
      "\n",
      "Copying Arrow Files (14 of 293)\n",
      "\n",
      "Copying Arrow Files (15 of 293)\n",
      "\n",
      "Copying Arrow Files (16 of 293)\n",
      "\n",
      "Copying Arrow Files (17 of 293)\n",
      "\n",
      "Copying Arrow Files (18 of 293)\n",
      "\n",
      "Copying Arrow Files (19 of 293)\n",
      "\n",
      "Copying Arrow Files (20 of 293)\n",
      "\n",
      "Copying Arrow Files (21 of 293)\n",
      "\n",
      "Copying Arrow Files (22 of 293)\n",
      "\n",
      "Copying Arrow Files (23 of 293)\n",
      "\n",
      "Copying Arrow Files (24 of 293)\n",
      "\n",
      "Copying Arrow Files (25 of 293)\n",
      "\n",
      "Copying Arrow Files (26 of 293)\n",
      "\n",
      "Copying Arrow Files (27 of 293)\n",
      "\n",
      "Copying Arrow Files (28 of 293)\n",
      "\n",
      "Copying Arrow Files (29 of 293)\n",
      "\n",
      "Copying Arrow Files (30 of 293)\n",
      "\n",
      "Copying Arrow Files (31 of 293)\n",
      "\n",
      "Copying Arrow Files (32 of 293)\n",
      "\n",
      "Copying Arrow Files (33 of 293)\n",
      "\n",
      "Copying Arrow Files (34 of 293)\n",
      "\n",
      "Copying Arrow Files (35 of 293)\n",
      "\n",
      "Copying Arrow Files (36 of 293)\n",
      "\n",
      "Copying Arrow Files (37 of 293)\n",
      "\n",
      "Copying Arrow Files (38 of 293)\n",
      "\n",
      "Copying Arrow Files (39 of 293)\n",
      "\n",
      "Copying Arrow Files (40 of 293)\n",
      "\n",
      "Copying Arrow Files (41 of 293)\n",
      "\n",
      "Copying Arrow Files (42 of 293)\n",
      "\n",
      "Copying Arrow Files (43 of 293)\n",
      "\n",
      "Copying Arrow Files (44 of 293)\n",
      "\n",
      "Copying Arrow Files (45 of 293)\n",
      "\n",
      "Copying Arrow Files (46 of 293)\n",
      "\n",
      "Copying Arrow Files (47 of 293)\n",
      "\n",
      "Copying Arrow Files (48 of 293)\n",
      "\n",
      "Copying Arrow Files (49 of 293)\n",
      "\n",
      "Copying Arrow Files (50 of 293)\n",
      "\n",
      "Copying Arrow Files (51 of 293)\n",
      "\n",
      "Copying Arrow Files (52 of 293)\n",
      "\n",
      "Copying Arrow Files (53 of 293)\n",
      "\n",
      "Copying Arrow Files (54 of 293)\n",
      "\n",
      "Copying Arrow Files (55 of 293)\n",
      "\n",
      "Copying Arrow Files (56 of 293)\n",
      "\n",
      "Copying Arrow Files (57 of 293)\n",
      "\n",
      "Copying Arrow Files (58 of 293)\n",
      "\n",
      "Copying Arrow Files (59 of 293)\n",
      "\n",
      "Copying Arrow Files (60 of 293)\n",
      "\n",
      "Copying Arrow Files (61 of 293)\n",
      "\n",
      "Copying Arrow Files (62 of 293)\n",
      "\n",
      "Copying Arrow Files (63 of 293)\n",
      "\n",
      "Copying Arrow Files (64 of 293)\n",
      "\n",
      "Copying Arrow Files (65 of 293)\n",
      "\n",
      "Copying Arrow Files (66 of 293)\n",
      "\n",
      "Copying Arrow Files (67 of 293)\n",
      "\n",
      "Copying Arrow Files (68 of 293)\n",
      "\n",
      "Copying Arrow Files (69 of 293)\n",
      "\n",
      "Copying Arrow Files (70 of 293)\n",
      "\n",
      "Copying Arrow Files (71 of 293)\n",
      "\n",
      "Copying Arrow Files (72 of 293)\n",
      "\n",
      "Copying Arrow Files (73 of 293)\n",
      "\n",
      "Copying Arrow Files (74 of 293)\n",
      "\n",
      "Copying Arrow Files (75 of 293)\n",
      "\n",
      "Copying Arrow Files (76 of 293)\n",
      "\n",
      "Copying Arrow Files (77 of 293)\n",
      "\n",
      "Copying Arrow Files (78 of 293)\n",
      "\n",
      "Copying Arrow Files (79 of 293)\n",
      "\n",
      "Copying Arrow Files (80 of 293)\n",
      "\n",
      "Copying Arrow Files (81 of 293)\n",
      "\n",
      "Copying Arrow Files (82 of 293)\n",
      "\n",
      "Copying Arrow Files (83 of 293)\n",
      "\n",
      "Copying Arrow Files (84 of 293)\n",
      "\n",
      "Copying Arrow Files (85 of 293)\n",
      "\n",
      "Copying Arrow Files (86 of 293)\n",
      "\n",
      "Copying Arrow Files (87 of 293)\n",
      "\n",
      "Copying Arrow Files (88 of 293)\n",
      "\n",
      "Copying Arrow Files (89 of 293)\n",
      "\n",
      "Copying Arrow Files (90 of 293)\n",
      "\n",
      "Copying Arrow Files (91 of 293)\n",
      "\n",
      "Copying Arrow Files (92 of 293)\n",
      "\n",
      "Copying Arrow Files (93 of 293)\n",
      "\n",
      "Copying Arrow Files (94 of 293)\n",
      "\n",
      "Copying Arrow Files (95 of 293)\n",
      "\n",
      "Copying Arrow Files (96 of 293)\n",
      "\n",
      "Copying Arrow Files (97 of 293)\n",
      "\n",
      "Copying Arrow Files (98 of 293)\n",
      "\n",
      "Copying Arrow Files (99 of 293)\n",
      "\n",
      "Copying Arrow Files (100 of 293)\n",
      "\n",
      "Copying Arrow Files (101 of 293)\n",
      "\n",
      "Copying Arrow Files (102 of 293)\n",
      "\n",
      "Copying Arrow Files (103 of 293)\n",
      "\n",
      "Copying Arrow Files (104 of 293)\n",
      "\n",
      "Copying Arrow Files (105 of 293)\n",
      "\n",
      "Copying Arrow Files (106 of 293)\n",
      "\n",
      "Copying Arrow Files (107 of 293)\n",
      "\n",
      "Copying Arrow Files (108 of 293)\n",
      "\n",
      "Copying Arrow Files (109 of 293)\n",
      "\n",
      "Copying Arrow Files (110 of 293)\n",
      "\n",
      "Copying Arrow Files (111 of 293)\n",
      "\n",
      "Copying Arrow Files (112 of 293)\n",
      "\n",
      "Copying Arrow Files (113 of 293)\n",
      "\n",
      "Copying Arrow Files (114 of 293)\n",
      "\n",
      "Copying Arrow Files (115 of 293)\n",
      "\n",
      "Copying Arrow Files (116 of 293)\n",
      "\n",
      "Copying Arrow Files (117 of 293)\n",
      "\n",
      "Copying Arrow Files (118 of 293)\n",
      "\n",
      "Copying Arrow Files (119 of 293)\n",
      "\n",
      "Copying Arrow Files (120 of 293)\n",
      "\n",
      "Copying Arrow Files (121 of 293)\n",
      "\n",
      "Copying Arrow Files (122 of 293)\n",
      "\n",
      "Copying Arrow Files (123 of 293)\n",
      "\n",
      "Copying Arrow Files (124 of 293)\n",
      "\n",
      "Copying Arrow Files (125 of 293)\n",
      "\n",
      "Copying Arrow Files (126 of 293)\n",
      "\n",
      "Copying Arrow Files (127 of 293)\n",
      "\n",
      "Copying Arrow Files (128 of 293)\n",
      "\n",
      "Copying Arrow Files (129 of 293)\n",
      "\n",
      "Copying Arrow Files (130 of 293)\n",
      "\n",
      "Copying Arrow Files (131 of 293)\n",
      "\n",
      "Copying Arrow Files (132 of 293)\n",
      "\n",
      "Copying Arrow Files (133 of 293)\n",
      "\n",
      "Copying Arrow Files (134 of 293)\n",
      "\n",
      "Copying Arrow Files (135 of 293)\n",
      "\n",
      "Copying Arrow Files (136 of 293)\n",
      "\n",
      "Copying Arrow Files (137 of 293)\n",
      "\n",
      "Copying Arrow Files (138 of 293)\n",
      "\n",
      "Copying Arrow Files (139 of 293)\n",
      "\n",
      "Copying Arrow Files (140 of 293)\n",
      "\n",
      "Copying Arrow Files (141 of 293)\n",
      "\n",
      "Copying Arrow Files (142 of 293)\n",
      "\n",
      "Copying Arrow Files (143 of 293)\n",
      "\n",
      "Copying Arrow Files (144 of 293)\n",
      "\n",
      "Copying Arrow Files (145 of 293)\n",
      "\n",
      "Copying Arrow Files (146 of 293)\n",
      "\n",
      "Copying Arrow Files (147 of 293)\n",
      "\n",
      "Copying Arrow Files (148 of 293)\n",
      "\n",
      "Copying Arrow Files (149 of 293)\n",
      "\n",
      "Copying Arrow Files (150 of 293)\n",
      "\n",
      "Copying Arrow Files (151 of 293)\n",
      "\n",
      "Copying Arrow Files (152 of 293)\n",
      "\n",
      "Copying Arrow Files (153 of 293)\n",
      "\n",
      "Copying Arrow Files (154 of 293)\n",
      "\n",
      "Copying Arrow Files (155 of 293)\n",
      "\n",
      "Copying Arrow Files (156 of 293)\n",
      "\n",
      "Copying Arrow Files (157 of 293)\n",
      "\n",
      "Copying Arrow Files (158 of 293)\n",
      "\n",
      "Copying Arrow Files (159 of 293)\n",
      "\n",
      "Copying Arrow Files (160 of 293)\n",
      "\n",
      "Copying Arrow Files (161 of 293)\n",
      "\n",
      "Copying Arrow Files (162 of 293)\n",
      "\n",
      "Copying Arrow Files (163 of 293)\n",
      "\n",
      "Copying Arrow Files (164 of 293)\n",
      "\n",
      "Copying Arrow Files (165 of 293)\n",
      "\n",
      "Copying Arrow Files (166 of 293)\n",
      "\n",
      "Copying Arrow Files (167 of 293)\n",
      "\n",
      "Copying Arrow Files (168 of 293)\n",
      "\n",
      "Copying Arrow Files (169 of 293)\n",
      "\n",
      "Copying Arrow Files (170 of 293)\n",
      "\n",
      "Copying Arrow Files (171 of 293)\n",
      "\n",
      "Copying Arrow Files (172 of 293)\n",
      "\n",
      "Copying Arrow Files (173 of 293)\n",
      "\n",
      "Copying Arrow Files (174 of 293)\n",
      "\n",
      "Copying Arrow Files (175 of 293)\n",
      "\n",
      "Copying Arrow Files (176 of 293)\n",
      "\n",
      "Copying Arrow Files (177 of 293)\n",
      "\n",
      "Copying Arrow Files (178 of 293)\n",
      "\n",
      "Copying Arrow Files (179 of 293)\n",
      "\n",
      "Copying Arrow Files (180 of 293)\n",
      "\n",
      "Copying Arrow Files (181 of 293)\n",
      "\n",
      "Copying Arrow Files (182 of 293)\n",
      "\n",
      "Copying Arrow Files (183 of 293)\n",
      "\n",
      "Copying Arrow Files (184 of 293)\n",
      "\n",
      "Copying Arrow Files (185 of 293)\n",
      "\n",
      "Copying Arrow Files (186 of 293)\n",
      "\n",
      "Copying Arrow Files (187 of 293)\n",
      "\n",
      "Copying Arrow Files (188 of 293)\n",
      "\n",
      "Copying Arrow Files (189 of 293)\n",
      "\n",
      "Copying Arrow Files (190 of 293)\n",
      "\n",
      "Copying Arrow Files (191 of 293)\n",
      "\n",
      "Copying Arrow Files (192 of 293)\n",
      "\n",
      "Copying Arrow Files (193 of 293)\n",
      "\n",
      "Copying Arrow Files (194 of 293)\n",
      "\n",
      "Copying Arrow Files (195 of 293)\n",
      "\n",
      "Copying Arrow Files (196 of 293)\n",
      "\n",
      "Copying Arrow Files (197 of 293)\n",
      "\n",
      "Copying Arrow Files (198 of 293)\n",
      "\n",
      "Copying Arrow Files (199 of 293)\n",
      "\n",
      "Copying Arrow Files (200 of 293)\n",
      "\n",
      "Copying Arrow Files (201 of 293)\n",
      "\n",
      "Copying Arrow Files (202 of 293)\n",
      "\n",
      "Copying Arrow Files (203 of 293)\n",
      "\n",
      "Copying Arrow Files (204 of 293)\n",
      "\n",
      "Copying Arrow Files (205 of 293)\n",
      "\n",
      "Copying Arrow Files (206 of 293)\n",
      "\n",
      "Copying Arrow Files (207 of 293)\n",
      "\n",
      "Copying Arrow Files (208 of 293)\n",
      "\n",
      "Copying Arrow Files (209 of 293)\n",
      "\n",
      "Copying Arrow Files (210 of 293)\n",
      "\n",
      "Copying Arrow Files (211 of 293)\n",
      "\n",
      "Copying Arrow Files (212 of 293)\n",
      "\n",
      "Copying Arrow Files (213 of 293)\n",
      "\n",
      "Copying Arrow Files (214 of 293)\n",
      "\n",
      "Copying Arrow Files (215 of 293)\n",
      "\n",
      "Copying Arrow Files (216 of 293)\n",
      "\n",
      "Copying Arrow Files (217 of 293)\n",
      "\n",
      "Copying Arrow Files (218 of 293)\n",
      "\n",
      "Copying Arrow Files (219 of 293)\n",
      "\n",
      "Copying Arrow Files (220 of 293)\n",
      "\n",
      "Copying Arrow Files (221 of 293)\n",
      "\n",
      "Copying Arrow Files (222 of 293)\n",
      "\n",
      "Copying Arrow Files (223 of 293)\n",
      "\n",
      "Copying Arrow Files (224 of 293)\n",
      "\n",
      "Copying Arrow Files (225 of 293)\n",
      "\n",
      "Copying Arrow Files (226 of 293)\n",
      "\n",
      "Copying Arrow Files (227 of 293)\n",
      "\n",
      "Copying Arrow Files (228 of 293)\n",
      "\n",
      "Copying Arrow Files (229 of 293)\n",
      "\n",
      "Copying Arrow Files (230 of 293)\n",
      "\n",
      "Copying Arrow Files (231 of 293)\n",
      "\n",
      "Copying Arrow Files (232 of 293)\n",
      "\n",
      "Copying Arrow Files (233 of 293)\n",
      "\n",
      "Copying Arrow Files (234 of 293)\n",
      "\n",
      "Copying Arrow Files (235 of 293)\n",
      "\n",
      "Copying Arrow Files (236 of 293)\n",
      "\n",
      "Copying Arrow Files (237 of 293)\n",
      "\n",
      "Copying Arrow Files (238 of 293)\n",
      "\n",
      "Copying Arrow Files (239 of 293)\n",
      "\n",
      "Copying Arrow Files (240 of 293)\n",
      "\n",
      "Copying Arrow Files (241 of 293)\n",
      "\n",
      "Copying Arrow Files (242 of 293)\n",
      "\n",
      "Copying Arrow Files (243 of 293)\n",
      "\n",
      "Copying Arrow Files (244 of 293)\n",
      "\n",
      "Copying Arrow Files (245 of 293)\n",
      "\n",
      "Copying Arrow Files (246 of 293)\n",
      "\n",
      "Copying Arrow Files (247 of 293)\n",
      "\n",
      "Copying Arrow Files (248 of 293)\n",
      "\n",
      "Copying Arrow Files (249 of 293)\n",
      "\n",
      "Copying Arrow Files (250 of 293)\n",
      "\n",
      "Copying Arrow Files (251 of 293)\n",
      "\n",
      "Copying Arrow Files (252 of 293)\n",
      "\n",
      "Copying Arrow Files (253 of 293)\n",
      "\n",
      "Copying Arrow Files (254 of 293)\n",
      "\n",
      "Copying Arrow Files (255 of 293)\n",
      "\n",
      "Copying Arrow Files (256 of 293)\n",
      "\n",
      "Copying Arrow Files (257 of 293)\n",
      "\n",
      "Copying Arrow Files (258 of 293)\n",
      "\n",
      "Copying Arrow Files (259 of 293)\n",
      "\n",
      "Copying Arrow Files (260 of 293)\n",
      "\n",
      "Copying Arrow Files (261 of 293)\n",
      "\n",
      "Copying Arrow Files (262 of 293)\n",
      "\n",
      "Copying Arrow Files (263 of 293)\n",
      "\n",
      "Copying Arrow Files (264 of 293)\n",
      "\n",
      "Copying Arrow Files (265 of 293)\n",
      "\n",
      "Copying Arrow Files (266 of 293)\n",
      "\n",
      "Copying Arrow Files (267 of 293)\n",
      "\n",
      "Copying Arrow Files (268 of 293)\n",
      "\n",
      "Copying Arrow Files (269 of 293)\n",
      "\n",
      "Copying Arrow Files (270 of 293)\n",
      "\n",
      "Copying Arrow Files (271 of 293)\n",
      "\n",
      "Copying Arrow Files (272 of 293)\n",
      "\n",
      "Copying Arrow Files (273 of 293)\n",
      "\n",
      "Copying Arrow Files (274 of 293)\n",
      "\n",
      "Copying Arrow Files (275 of 293)\n",
      "\n",
      "Copying Arrow Files (276 of 293)\n",
      "\n",
      "Copying Arrow Files (277 of 293)\n",
      "\n",
      "Copying Arrow Files (278 of 293)\n",
      "\n",
      "Copying Arrow Files (279 of 293)\n",
      "\n",
      "Copying Arrow Files (280 of 293)\n",
      "\n",
      "Copying Arrow Files (281 of 293)\n",
      "\n",
      "Copying Arrow Files (282 of 293)\n",
      "\n",
      "Copying Arrow Files (283 of 293)\n",
      "\n",
      "Copying Arrow Files (284 of 293)\n",
      "\n",
      "Copying Arrow Files (285 of 293)\n",
      "\n",
      "Copying Arrow Files (286 of 293)\n",
      "\n",
      "Copying Arrow Files (287 of 293)\n",
      "\n",
      "Copying Arrow Files (288 of 293)\n",
      "\n",
      "Copying Arrow Files (289 of 293)\n",
      "\n",
      "Copying Arrow Files (290 of 293)\n",
      "\n",
      "Copying Arrow Files (291 of 293)\n",
      "\n",
      "Copying Arrow Files (292 of 293)\n",
      "\n",
      "Copying Arrow Files (293 of 293)\n",
      "\n",
      "Getting ImputeWeights\n",
      "\n",
      "No imputeWeights found, returning NULL\n",
      "\n",
      "Copying Other Files...\n",
      "\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in saveArchRProject(ArchRProj = atac, outputDirectory = \"/home/icb/zihe.zheng/projects/microglia/data/SUN/arrows/\", : all(file.exists(zfiles)) is not TRUE\n",
     "output_type": "error",
     "traceback": [
      "Error in saveArchRProject(ArchRProj = atac, outputDirectory = \"/home/icb/zihe.zheng/projects/microglia/data/SUN/arrows/\", : all(file.exists(zfiles)) is not TRUE\nTraceback:\n",
      "1. saveArchRProject(ArchRProj = atac, outputDirectory = \"/home/icb/zihe.zheng/projects/microglia/data/SUN/arrows/\", \n .     load = FALSE)",
      "2. stopifnot(all(file.exists(zfiles)))"
     ]
    }
   ],
   "source": [
    "saveArchRProject(ArchRProj = atac, outputDirectory = \"/home/icb/zihe.zheng/projects/microglia/data/SUN/arrows/\", load = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4407573b-439e-49a9-874c-317bc0e5a0fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "loadArchRProject(path = \"/home/icb/zihe.zheng/projects/microglia/data/SUN/microglia.snATAC.ArchRobj.rds\", force = FALSE, showLogo = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "bf0ee5e7-04ed-49cd-9f43-ed120dace4db",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'atac.ArrowFiles' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'atac.ArrowFiles' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "atac.ArrowFiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "db62a8b0-9d5b-42d7-af7d-e67beacfb9bf",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in h5checktypeOrOpenLoc(file, readonly = TRUE, fapl = NULL, native = native): Error in h5checktypeOrOpenLoc(). Cannot open file. File 'D19-122585.arrow' does not exist.\n",
     "output_type": "error",
     "traceback": [
      "Error in h5checktypeOrOpenLoc(file, readonly = TRUE, fapl = NULL, native = native): Error in h5checktypeOrOpenLoc(). Cannot open file. File 'D19-122585.arrow' does not exist.\nTraceback:\n",
      "1. getAvailableMatrices(atac)",
      "2. .availableArrays(getArrowFiles(ArchRProj = ArchRProj))",
      "3. .safelapply(seq_along(ArrowFiles), function(x) {\n .     groups <- h5ls(ArrowFiles[x]) %>% {\n .         .[.$group == \"/\" & .$otype == \"H5I_GROUP\", \"name\"]\n .     }\n .     groups <- groups[!grepl(\"Fragments|Metadata\", groups)]\n .     groups\n . }, threads = threads) %>% Reduce(\"intersect\", .)",
      "4. Reduce(\"intersect\", .)",
      "5. .safelapply(seq_along(ArrowFiles), function(x) {\n .     groups <- h5ls(ArrowFiles[x]) %>% {\n .         .[.$group == \"/\" & .$otype == \"H5I_GROUP\", \"name\"]\n .     }\n .     groups <- groups[!grepl(\"Fragments|Metadata\", groups)]\n .     groups\n . }, threads = threads)",
      "6. lapply(...)",
      "7. FUN(X[[i]], ...)",
      "8. h5ls(ArrowFiles[x]) %>% {\n .     .[.$group == \"/\" & .$otype == \"H5I_GROUP\", \"name\"]\n . }",
      "9. h5ls(ArrowFiles[x])",
      "10. h5checktypeOrOpenLoc(file, readonly = TRUE, fapl = NULL, native = native)",
      "11. stop(\"Error in \", fctname, \". Cannot open file. File '\", file, \n  .     \"' does not exist.\")"
     ]
    }
   ],
   "source": [
    "getAvailableMatrices(atac)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "121444d1-3352-48ad-8996-6cce3d4f9673",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in value[[3L]](cond): select Not Found in Colnames of cellColData:\n\u001b[1m\u001b[33mError\u001b[39m in `dplyr::mutate()`:\u001b[22m\n\u001b[1m\u001b[22m\u001b[36mℹ\u001b[39m In argument: `tmpNewCol123 = eval(parse(text = select[x]))`.\n\u001b[1mCaused by error:\u001b[22m\n\u001b[33m!\u001b[39m object 'DoubletEnrichment' not found\n\n",
     "output_type": "error",
     "traceback": [
      "Error in value[[3L]](cond): select Not Found in Colnames of cellColData:\n\u001b[1m\u001b[33mError\u001b[39m in `dplyr::mutate()`:\u001b[22m\n\u001b[1m\u001b[22m\u001b[36mℹ\u001b[39m In argument: `tmpNewCol123 = eval(parse(text = select[x]))`.\n\u001b[1mCaused by error:\u001b[22m\n\u001b[33m!\u001b[39m object 'DoubletEnrichment' not found\n\nTraceback:\n",
      "1. filterDoublets(ArchRProj = atac)",
      "2. getCellColData(ArchRProj, c(\"Sample\", \"DoubletEnrichment\", \"DoubletScore\"))",
      "3. lapply(seq_along(select), function(x) {\n .     tryCatch({\n .         data.frame(dplyr::mutate(ccd, tmpNewCol123 = eval(parse(text = select[x])))[, \n .             \"tmpNewCol123\"], stringsAsFactors = FALSE)\n .     }, error = function(x) {\n .         stop(\"select Not Found in Colnames of cellColData:\\n\", \n .             x)\n .     })\n . }) %>% Reduce(\"cbind\", .) %>% {\n .     data.frame(., stringsAsFactors = FALSE)\n . }",
      "4. Reduce(\"cbind\", .)",
      "5. lapply(seq_along(select), function(x) {\n .     tryCatch({\n .         data.frame(dplyr::mutate(ccd, tmpNewCol123 = eval(parse(text = select[x])))[, \n .             \"tmpNewCol123\"], stringsAsFactors = FALSE)\n .     }, error = function(x) {\n .         stop(\"select Not Found in Colnames of cellColData:\\n\", \n .             x)\n .     })\n . })",
      "6. FUN(X[[i]], ...)",
      "7. tryCatch({\n .     data.frame(dplyr::mutate(ccd, tmpNewCol123 = eval(parse(text = select[x])))[, \n .         \"tmpNewCol123\"], stringsAsFactors = FALSE)\n . }, error = function(x) {\n .     stop(\"select Not Found in Colnames of cellColData:\\n\", x)\n . })",
      "8. tryCatchList(expr, classes, parentenv, handlers)",
      "9. tryCatchOne(expr, names, parentenv, handlers[[1L]])",
      "10. value[[3L]](cond)",
      "11. stop(\"select Not Found in Colnames of cellColData:\\n\", x)"
     ]
    }
   ],
   "source": [
    "proj <- filterDoublets(ArchRProj = atac)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83a7d4ff-4ee8-4c1d-86d9-1736c67154a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# there is no actual data in the rds file for atac-seq"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "r-archr",
   "language": "R",
   "name": "r-archr"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
